File: code/jvm/src/main/kotlin/pt/isel/leic/ptgest/domain/workout/model/SetExerciseDetails.kt
package pt.isel.leic.ptgest.domain.workout.model

import pt.isel.leic.ptgest.domain.workout.Modality
import pt.isel.leic.ptgest.domain.workout.MuscleGroup

data class SetExerciseDetails(
    val id: Int,
    val name: String,
    val muscleGroup: List<MuscleGroup>,
    val modality: Modality,
    val details: Map<String, Any>
)

File: code/js/src/views/user/trainerViews/models/sets/SetExerciseDetails.ts
class SetExerciseDetails {
    id: number
    name: string
    muscleGroup: string[] | null
    modality: string | null
    details: Map<string, any>

    constructor(id: number, name: string, muscleGroup: string[] | null, modality: string | null, details: Map<string, any>) {
        this.id = id
        this.name = name
        this.muscleGroup = muscleGroup
        this.modality = modality
        this.details = details
    }
}
export default SetExerciseDetails
